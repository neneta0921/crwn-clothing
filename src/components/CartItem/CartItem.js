import {
  CartItemContainer,
  CartItemImage,
  ItemDetailsContainer
} from './CartItem.styles';

export const CartItem = ({ item: { imageUrl, price, name, quantity } }) => (
  <CartItemContainer>
    <CartItemImage src={imageUrl} alt="item"/>
    <ItemDetailsContainer>
      <span className='name'>{name}</span>
      <span className='price'>
        {quantity} x ${price}
      </span>
    </ItemDetailsContainer>
  </CartItemContainer>
)
