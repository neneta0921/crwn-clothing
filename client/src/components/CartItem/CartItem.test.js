import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { CartItem } from './CartItem';

it('should render cartItem component', () => {
  const mockItem = {
    imageUrl: 'www.testImage.com',
    price: 10,
    name: 'hats',
    quantity: 3,
  };

  const tree = renderer.create(shallow(<CartItem item={mockItem} />)).toJSON();
  expect(tree).toMatchSnapshot();
});
