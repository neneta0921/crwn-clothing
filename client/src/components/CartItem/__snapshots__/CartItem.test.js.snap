// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render cartItem component 1`] = `
<div
  className="sc-bdfBwQ bTXWnC"
>
  <img
    alt="item"
    className="sc-gsTCUz ebGfmt"
    src="www.testImage.com"
  />
  <div
    className="sc-dlfnbm iMlZdi"
  >
    <span>
      hats
    </span>
    <span>
      3
       x $
      10
    </span>
  </div>
</div>
`;
